### PR TITLE
Explanations for venus and euler

### PR DESCRIPTION
**Pull request by mviswanathsai**

### CODE REVIEW

**Comment by RobinNagpal** [comment]
For Venus Here are the comments
- We should explain the following features which we have not touched upon 
  - vaults
  - Distribution of XVS by assets
- Also the text needs to be updated which we can discuss int he call

File: generated/explanations/euler.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+This is the course header. This will be added on top of every page. Go to [DoDAO.io](https://www.dodao.io) to know more.
+
+ ---
+ 
+ ## Euler
+ 
+ **About Euler**        
+
+### What is Euler 
+Euler is a non-custodial DeFi protocol that allows users to lend and borrow and the main factor that differentiates Euler form other trusted protocols like Aave and Compound is that,  in Euler any asset that has a WETH pair on Uniswap v3 can be added as a lending market on Euler by anyone straight away. There is a  significant unmet demand for lending and borrowing  the long tail of crypto assets which are not listed on Aave or Compound. Euler is managed by holders of a protocol native governance token called Euler Governance Token (EUL).
```
Comment (RobinNagpal) [inline generated/explanations/euler.md:11]: Please use peppertype for this paragraph and other also as the sentence formation can be improved quite a bit.

File: generated/explanations/euler.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+### Asset Tiers
+Listing multiple assets which are not necessarily mainstream comes with risk as their price can fluctuate a lot. Euler mitigates this risk by using different types of  pools which can/cannot be isolated from other assets.  This limits the exposure in case there are any bad debts. Another concept adopted by Euler is that they consider the risk factor of both the collateral and the borrowed asset to come up with the maximum amount that can be borrowed for the said collateral-liability asset pair called **risk-adjusted collateral**.   For example, in borrowing XYZ with a collateral factor of 0.5, say you are using $1000 USDC with a collateral factor of 0.9, the maximum amount borrowable would come out ot be 1000*0.5*0.9 = $450 worth of XYZ as opposed to the usual 1000*0.9 that we are used to.
```
Comment (RobinNagpal) [inline generated/explanations/euler.md:14]: Please add `risk-adjusted collateral` as a separate point as its not too much related to Asset Tiers

File: generated/explanations/euler.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+* Isolation-tier -** ** assets are available for ordinary lending and borrowing, but they cannot be used as collateral to borrow other assets, and they can only be borrowed in isolation. What this means is that they cannot be borrowed alongside other assets using the same pool of collateral.  * Cross-tier -** ** assets are available for ordinary lending and borrowing, and cannot be used as collateral to borrow other assets, but they can be borrowed alongside other assets. For example, if a user has USDC and DAI as collateral, and they want to borrow cross-tier assets ABC and XYZ, then they can do so from a single account on Euler. * Collateral-tier - assets are available for ordinary lending and borrowing, cross-borrowing, and they can be used as collateral. For example, a user can deposit collateral assets DAI and USDC, and use them to borrow collateral assets UNI and LINK, all from a single account.
```
Comment (RobinNagpal) [inline generated/explanations/euler.md:15]: Improve the formatting.

Do you cross check the formatting in the generated markdown files?

https://dodao.academy/course/file/view/lending-and-borrowing-in-defi/euler/explanation/about-euler

File: generated/explanations/euler.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
...djusted collateral. A borrower who has just become in violation still has enough collateral to repay their loan, but is at risk of defaulting on their loan. As a result,  they may be liquidated to limit the potential for them to default. One of the issues with a fixed discount percentage for the liquidators on the collateral of the loan they repaid is that potential liquidators often have to engage in priority gas auctions (PGA) in order to find profitable liquidations, which exposes  the liquidation bonus as MEV (miner extractable value). In Euler they allow the discount to increase depending on how far under water a position is. Now, liquidators are forced to engage into a type of Dutch auction. As the discount gradually goes up, each liquidator must decide if they want to bid for a liquidation at the current discount being offered. Liquidator A might  be able to profit at 4%, but liquidator B might be so at 3.5% because they have a more efficient operation. This process is made more efficient by the TWAP oracles used on the Euler network which allow for a more smooth price movement over time which facilitates the dutch auction, preventing all liquidators from jumping in at once.
```
Comment (RobinNagpal) [inline generated/explanations/euler.md:18]: Please format liquidation text into 2-3 paragraphs

File: generated/explanations/euler.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+#### Interest Rates
+To improve capital efficiency and do away with errors in determining parameters for the interest rates of every lending market, Euler uses reactive interest rates. By using a PID controller, Euler amplifies or dampens the rate of change in interest rates when utilization is above or below the target level of utilization respectively. As a result, reactive interest rates  that adapt to market conditions for the underlying asset in real-time are created, without the need for ongoing governance intervention. Interest is accrued every second on Euler.
+
+ 
+ **EUL token**        
+
+## EUL
+The total supply of EUL is 27,182,818 (in homage to Euler’s number,[ e](https://en.wikipedia.org/wiki/E_(mathematical_constant))). The total supply of EUL is fixed for the first 4 years, after which EUL token holders may enact a governance proposal to inflate the supply by a maximum 2.718% per year. In that scenario, newly minted EUL will enter circulation via the Treasury.
```
Comment (RobinNagpal) [inline generated/explanations/euler.md:46]: Please check the formatting for the EUL token step

https://dodao.academy/course/file/view/lending-and-borrowing-in-defi/euler/explanation/eul-Token

also the keys should be all small case `eul-Token` -> `eul-token`

File: generated/explanations/euler.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+The Euler protocol will be managed by holders of the Euler Governance Token (EUL), a native governance token for the protocol. EUL tokens will represent the voting power of the protocol software. Holders with enough EUL tokens will be able to make a formal proposal for change on the protocol. Token holders will then be able to vote on the proposal themselves or delegate their vote shares to a third party. 
+Decisions token holders might vote on include proposals to alter:
+* The tier of an asset * Collateral and borrow factors * Price oracle parameters * Reactive interest rate model parameters * Reserve factors
+**How can you earn EUL?** The Euler token (EUL) is distributed to borrowers on select markets on the platform. Note that EUL is only earned by borrowing and cannot be earned by simply lending to the protocol.
+
+   
+  
+ **References**        
+Here are some important links to learn more about Compound III (V3)
```
Comment (RobinNagpal) [inline generated/explanations/euler.md:55]: What about link to the docs?

File: src/questions/venus.yaml [hunk 7]
```diff
@@ -0,0 +1,229 @@
+- uuid: 9e285187-2ae0-42dc-8b91-e36926634f47
+  type: SingleChoice
+  content: Which of these features are available in Venus?
+  hint: NoHint
+  explanation: Venus has the features of a money market protocol and stablecoin minting protocol together.
+  answerKeys:
+    - D
+  difficultyLevel: "Low"
+  choices:
+    - content: Money market features
```
Comment (RobinNagpal) [inline src/questions/venus.yaml:10]: `Money market features` -> `Lending and Borrowing`

File: src/questions/venus.yaml [hunk 8]
```diff
@@ -0,0 +1,229 @@
...
+    - content: To accrue interest on assets supplied and be used as collateral
+      key: A
+      
+    - content: To accrue debt and be used as collateral
+      key: B
+      
+    - content: To accrue debt only
+      key: C
+      
+    - content: To accrue debt and interest for borrowed and supplied assets respectively
```
Comment (RobinNagpal) [inline src/questions/venus.yaml:40]: Is D a wrong option?

File: src/questions/venus.yaml [hunk 9]
```diff
@@ -0,0 +1,229 @@
...
+- uuid: 6d862849-fcb2-4d76-b96c-170f556d5c72
+  type: SingleChoice
+  content: The Mint function in Venus is used to?
+  hint: NoHint
+  explanation: The mint function in venus is used to supply assets to the protocol and mint an equivalent amount of vTokens for the underlying asset.
+  answerKeys:
+    - C
+  difficultyLevel: "Medium"
+  choices:
+    - content: To mint their stable coin, VAI
```
Comment (RobinNagpal) [inline src/questions/venus.yaml:52]: Is A a wrong option

File: src/questions/venus.yaml [hunk 10]
```diff
@@ -0,0 +1,229 @@
...
+    - content: VRT is Venus Reward Token, used to gain interest at a fixed rate
+      key: C
+      
+    - content: VRT is Venus Return Token, used to redeem XVS at a fixed rate
+      key: D
+      
+
+
+- uuid: 87d32f53-099d-4188-89a1-e942d7d408e3
+  type: SingleChoice
```
Comment (RobinNagpal) [inline src/questions/venus.yaml:174]: Not a good question

File: src/questions/venus.yaml [hunk 11]
```diff
@@ -0,0 +1,229 @@
...
+    - B
+  difficultyLevel: "High"
+  choices:
+    - content: There are no vaults in Venus
+      key: A
+      
+    - content: In case Venus risk funds are not sufficient to cover the shortfall, the XVS vault will be liquidated to to a maximum of 10% per user's deposited XVS to cover the loss
+      key: B
+      
+    - content: Immediately, the XVS vault will be liquidated to to a maximum of 10% per user's deposited XVS to cover the loss
```
Comment (RobinNagpal) [inline src/questions/venus.yaml:203]: Is C a wrong option?

File: src/questions/venus.yaml (hunk 9 above)
Comment (mviswanathsai) [reply src/questions/venus.yaml:52]: Mint (https://docs.venus.io/docs/vtokens#mint)
The mint function transfers an asset into the protocol, which begins accumulating interest based on the current 
Supply Rate for The user receives a quantity of vTokens equal to the underlying tokens tokens supplied, divided by the current [Exchange Rate](https://docs.venus.io/docs/vtokens#vtokens-exchange-rate).

File: src/questions/venus.yaml (hunk 11 above)
Comment (mviswanathsai) [reply src/questions/venus.yaml:203]: yes, only when the risk funds are not sufficient to cover the losses, the xvs vault will be liquidated. Not immediately when a shortfall occurs

File: src/questions/venus.yaml (hunk 9 above)
Comment (mviswanathsai) [reply src/questions/venus.yaml:52]: VAI is minted used mintVAI function

**Pull request merged**